test/consistent-package-json.test.ts

```diff
@@ -0,0 +1,133 @@
+import {dirname, join, relative, resolve} from 'path';
+
+import {readFileSync, readJSONSync} from 'fs-extra';
+import glob from 'glob';
+
+const KNOWN_TEMPLATE_KEYS = [
+  'author',
+  'bugs',
+  'dependencies',
+  'description',
+  'files',
+  'homepage',
+  'license',
+  'main',
+  'name',
+  'publishConfig',
+  'repository',
+  'scripts',
+  'sideEffects',
+  'types',
+  'version',
+];
+
+const ROOT_PATH = resolve(__dirname, '..');
+const packageJSONTemplatePath = join('templates', 'package.hbs.json');
+const rawPackageJSONTemplate = readFileSync(
+  join(ROOT_PATH, packageJSONTemplatePath),
+  {encoding: 'utf8'},
+);
+const PACKAGE_NAME_PATTERN = /{{name}}/g;
+
+// eslint-disable-next-line jest/valid-describe
+describe(packageJSONTemplatePath, () => {
+  it('has only known keys', () => {
+    const keys = Object.keys(JSON.parse(rawPackageJSONTemplate)).sort();
+
+    // Template keys should match exactly. If updating it, remember to update tests!
+    expect(keys).toStrictEqual(KNOWN_TEMPLATE_KEYS);
+  });
+});
+
+readPackages().forEach(({packageName, packageJSON, packageJSONPath}) => {
+  const expectedPackageJSON = compileTemplate();
+
+  // eslint-disable-next-line jest/valid-describe
+  describe(packageJSONPath, () => {
```

Comment:
I get the intention of making a comprehensive suite that makes pinpointing errors very precise...  but the end result is a ton of output to sift through when I'm running the full suite (or trying to sift through CI logs).

I would prefer to see the test structure inverted, with each test proving a specific fact about all packages.  Yes, folks will have to grep a bit for error strings, but:
* The end result is less noisy during successful runs
* It's rare to touch >3 package.jsons in a PR, so the set of packages to sift through is small

-----

For context, here is a screencap of the test output with boilerplatey tests that focus on a single aspect of all packages:

<img width="586" alt="A Quilt test run showing a failing 'all packages link to Quilt issue tracker'-style test.  The test has somewhat oblique output, but it should be easy enough to grep the codebase for it." src="https://user-images.githubusercontent.com/673655/72318396-25648780-366a-11ea-961a-97606a5963bb.png">

Contrast with the current output at my terminal's lowest zoom setting:

<img width="110" alt="quilt 2020-01-14 00-49-59" src="https://user-images.githubusercontent.com/673655/72317604-cf8ee000-3667-11ea-87e6-1577a8b2616f.png">

That's ~70% of the output :/

Reply:
`compileTemplate` is fast.  Compiling lots of templates over and over again is fine if it makes test output more readable.